### PR TITLE
Support authentication on root custom resolver fields

### LINKED ISSUE #3746: GraphQLError: The directive "@authentication" can only be used once at this location.
**Describe the bug**
Getting an error when using `@authentication` for a specific query.

```
[
  GraphQLError: The directive "@authentication" can only be used once at this location.
      at Object.enter (/home/mathix/projects/peach_api/node_modules/graphql/validation/rules/UniqueDirectivesPerLocationRule.js:81:15)
      at Object.enter (/home/mathix/projects/peach_api/node_modules/graphql/language/visitor.js:301:32)
      at visit (/home/mathix/projects/peach_api/node_modules/graphql/language/visitor.js:197:21)
      at validateSDL (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/schema/validation/validate-sdl.ts:38:10)
      at validateUserDefinition (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/schema/validation/schema-validation.ts:106:31)
      at /home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/classes/Neo4jGraphQL.ts:334:39
      at new Promise (<anonymous>)
      at Neo4jGraphQL.generateExecutableSchema (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/classes/Neo4jGraphQL.ts:312:16)
      at Neo4jGraphQL.getExecutableSchema (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/classes/Neo4jGraphQL.ts:116:42)
      at Neo4jGraphQL.getSchema (/home/mathix/projects/peach_api/node_modules/@neo4j/graphql/src/classes/Neo4jGraphQL.ts:111:21) {
    path: undefined,
    locations: [ [Object] ],
    extensions: [Object: null prototype] {}
  }
]
```

**Type definitions**
```graphql
type Query {
    checkLogin: Boolean! @authentication
}
```

**To Reproduce**

1. Basic server following tutorial on Neo4j

Versions:
```
    "@apollo/server": "^4.9.1",
    "@neo4j/graphql": "^4.0.0-beta.1",
    "@neo4j/graphql-ogm": "^4.0.0-beta.1",
    "neo4j-driver": "^5.11.0",
    "graphql": "^16.7.1",
    "graphql-tag": "^2.12.6",
```

**Expected behavior**
Should do the authentication.

**System (please complete the following information):**

* OS: archlinux
* Version: @neo4j/graphql@4.0.0-beta.1 (also tested with latest v3)
* Node.js version: v18.16.1

### PR DESCRIPTION
**Pull request by a-alle**
# Description

Adds support for the `@authentication` directive on custom resolved fields of root types Query and Mutation.
Fixes #3746 

## Complexity

Complexity: Low

# Issue

> **Note**
>
>  Please link to the GitHub issue(s) in which the proposal for this work was discussed
>  
>  To link to multiple issues, use full syntax for each, for example `Closes #1, closes #2, closes #3`

Closes 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

File: .changeset/chilly-panthers-love.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@neo4j/graphql": patch
```
Comment (darrellwarde) [inline .changeset/chilly-panthers-love.md]: I'm thinking maybe this should be released in a minor version?

```suggestion
"@neo4j/graphql": minor
```

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

File: packages/graphql/src/classes/Neo4jGraphQL.ts [hunk 2]
```diff
@@ -284,23 +285,23 @@ class Neo4jGraphQL {
...
                 jwtPayloadFieldsMap: this.jwtFieldsMap,
-            });
+            };
+            subscriptionResolvers["Subscription.*"] = [wrapSubscription(wrapSubscriptionResolverArgs)];
         }
 
         // Merge generated and custom resolvers
         const mergedResolvers = mergeResolvers([...asArray(resolvers), ...asArray(this.resolvers)]);
-        return composeResolvers(mergedResolvers, resolversComposition);
+        return composeResolvers(mergedResolvers, { ...queryResolvers, ...mutationResolvers, ...subscriptionResolvers });
```
Comment (darrellwarde) [inline packages/graphql/src/classes/Neo4jGraphQL.ts]: It feels like this implementation is more complex than it needs to be. It seems like the `wrapQueryAndMutation` function is now differentiating between our generated resolvers and custom resolvers, and you have had to add a lot of metadata to the schema model to support this. But we already have this knowledge in this function. So we're merging the resolvers together just to then split them apart again. Happy to discuss, let me know if you want to talk it through!

**Review by darrellwarde — COMMENTED** [review]
Quite a high-level comment on this one, which as I say, I am happy to have a chat about - perhaps there is more complexity here that I'm not seeing!

**Review by darrellwarde — DISMISSED** [review]
Great, I think this is looking good!

**Pull request merged**